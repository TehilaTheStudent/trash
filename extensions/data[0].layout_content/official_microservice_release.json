[
  {
    "componentName": "d-form",
    "id": 1679043735519,
    "category": "form",
    "props": {
      "layout": "vertical",
      "ref": "formRef"
    },
    "children": [
      {
        "componentName": "d-form-item",
        "id": 1679043742334,
        "category": "form",
        "props": {
          "label": "Environment Level",
          "field": "categories",
          "help-tips": "",
          "visible": {
            "type": "fixed",
            "conditions": [],
            "value": true
          },
          "disabled": false,
          "rules": [
            {
              "required": true,
              "message": "This field cannot be left blank",
              "trigger": [
                "blur",
                "change"
              ]
            },
            {
              "regex": "",
              "message": "Invalid environment level",
              "trigger": [
                "blur",
                "change"
              ]
            }
          ]
        },
        "children": [
          {
            "componentName": "d-select",
            "id": 1679043742333,
            "props": {
              "selectType": "normal",
              "singleOptions": {
                "type": "api",
                "fixedOptions": [
                  {
                    "label": "选项",
                    "value": "1",
                    "defaultChecked": false,
                    "visible": false
                  }
                ],
                "apiOptions": {
                  "url": "/v2/${project_id}/environment/categories",
                  "method": "GET",
                  "params": {},
                  "header": {},
                  "label": "name",
                  "value": "id",
                  "body": {},
                  "responseUrl": "",
                  "linkedFields": [],
                  "remote": false,
                  "remoteName": "",
                  "remoteQueryField": "params"
                }
              },
              "isUpdateChildAll": true,
              "remote": false,
              "allow-clear": false,
              "filter": true,
              "disabled": {
                "type": "fixed",
                "conditions": [],
                "value": false
              }
            },
            "category": "form"
          }
        ]
      },
      {
        "componentName": "d-form-item",
        "id": 1715086422793,
        "category": "form",
        "props": {
          "label": "Environment",
          "field": "env_id",
          "help-tips": "Release the artfact based on the resources and strategies configured in environment of Release.",
          "rules": [
            {
              "required": true,
              "message": "Environment cannot be left blank.",
              "trigger": [
                "blur",
                "change"
              ]
            },
            {
              "regex": "",
              "message": "Environment Invalid content.",
              "trigger": [
                "blur",
                "change"
              ]
            }
          ],
          "visible": {
            "type": "fixed",
            "conditions": [],
            "value": true
          },
          "disabled": false
        },
        "children": [
          {
            "componentName": "link-select",
            "id": 1715086422792,
            "props": {
              "selectType": "normal",
              "singleOptions": {
                "type": "api",
                "fixedOptions": [
                  {
                    "label": "Option",
                    "value": "1",
                    "defaultChecked": false,
                    "visible": false
                  }
                ],
                "apiOptions": {
                  "url": "/v2/${project_id}/environments",
                  "method": "GET",
                  "params": {
                    "environment_category_id": "${categories}",
                    "offset": "0",
                    "limit": "200"
                  },
                  "header": {},
                  "label": "name",
                  "value": "id",
                  "body": {},
                  "responseUrl": "environments",
                  "linkedFields": [
                    "categories"
                  ],
                  "remote": false,
                  "remoteName": "",
                  "remoteQueryField": "params"
                }
              },
              "isUpdateChildAll": true,
              "remote": "",
              "allow-clear": false,
              "showPlaceholder": "",
              "filter": "",
              "linkText": "Create environment",
              "linkHref": "${origin}/cicd/project/${project_id}/environment?v=1",
              "herfTips": "",
              "isSubPipelineTag": false,
              "disabled": {
                "type": "fixed",
                "conditions": [],
                "value": false
              }
            },
            "category": "form"
          }
        ]
      },
      {
        "componentName": "d-form-item",
        "id": 1679043833875,
        "category": "form",
        "props": {
          "label": "Artifact Path",
          "field": "artifact",
          "help-tips": "",
          "visible": {
            "type": "fixed",
            "conditions": [],
            "value": true
          },
          "disabled": false,
          "rules": [
            {
              "required": true,
              "message": "This field cannot be left blank",
              "trigger": [
                "blur",
                "change"
              ]
            },
            {
              "regex": "^[\\s\\S]{1,500}$",
              "message": "It can contain a maximum of 500 characters.",
              "trigger": [
                "blur",
                "change"
              ]
            }
          ]
        },
        "children": [
          {
            "componentName": "d-input",
            "id": 1679043833874,
            "props": {
              "inputType": "normal",
              "placeholder": "请输入",
              "isTrim": true,
              "autocomplete": "off",
              "defaultValue": "",
              "disabled": {
                "type": "fixed",
                "conditions": [],
                "value": false
              }
            },
            "category": "form"
          }
        ]
      }
    ]
  }
]