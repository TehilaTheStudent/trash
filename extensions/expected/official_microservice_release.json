{
  "plugin_name": "official_microservice_release",
  "display_name": "",
  "data": [
    {
      "unique_id": "8758317c23d749de80537c7a5e1d9091",
      "name": "categories",
      "default_value": "",
      "plugin_name": "official_microservice_release",
      "version": "0.1.0",
      "type": "d-select",
      "workspace_id": "*",
      "layout_content": "[{\"componentName\":\"d-form\",\"id\":1679043735519,\"category\":\"form\",\"props\":{\"layout\":\"vertical\",\"ref\":\"formRef\"},\"children\":[{\"componentName\":\"d-form-item\",\"id\":1679043742334,\"category\":\"form\",\"props\":{\"label\":\"Environment Level\",\"field\":\"categories\",\"help-tips\":\"\",\"visible\":{\"type\":\"fixed\",\"conditions\":[],\"value\":true},\"disabled\":false,\"rules\":[{\"required\":true,\"message\":\"This field cannot be left blank\",\"trigger\":[\"blur\",\"change\"]},{\"regex\":\"\",\"message\":\"Invalid environment level\",\"trigger\":[\"blur\",\"change\"]}]},\"children\":[{\"componentName\":\"d-select\",\"id\":1679043742333,\"props\":{\"selectType\":\"normal\",\"singleOptions\":{\"type\":\"api\",\"fixedOptions\":[{\"label\":\"选项\",\"value\":\"1\",\"defaultChecked\":false,\"visible\":false}],\"apiOptions\":{\"url\":\"/v2/${project_id}/environment/categories\",\"method\":\"GET\",\"params\":{},\"header\":{},\"label\":\"name\",\"value\":\"id\",\"body\":{},\"responseUrl\":\"\",\"linkedFields\":[],\"remote\":false,\"remoteName\":\"\",\"remoteQueryField\":\"params\"}},\"isUpdateChildAll\":true,\"remote\":false,\"allow-clear\":false,\"filter\":true,\"disabled\":{\"type\":\"fixed\",\"conditions\":[],\"value\":false}},\"category\":\"form\"}]},{\"componentName\":\"d-form-item\",\"id\":1715086422793,\"category\":\"form\",\"props\":{\"label\":\"Environment\",\"field\":\"env_id\",\"help-tips\":\"Release the artfact based on the resources and strategies configured in environment of Release.\",\"rules\":[{\"required\":true,\"message\":\"Environment cannot be left blank.\",\"trigger\":[\"blur\",\"change\"]},{\"regex\":\"\",\"message\":\"Environment Invalid content.\",\"trigger\":[\"blur\",\"change\"]}],\"visible\":{\"type\":\"fixed\",\"conditions\":[],\"value\":true},\"disabled\":false},\"children\":[{\"componentName\":\"link-select\",\"id\":1715086422792,\"props\":{\"selectType\":\"normal\",\"singleOptions\":{\"type\":\"api\",\"fixedOptions\":[{\"label\":\"Option\",\"value\":\"1\",\"defaultChecked\":false,\"visible\":false}],\"apiOptions\":{\"url\":\"/v2/${project_id}/environments\",\"method\":\"GET\",\"params\":{\"environment_category_id\":\"${categories}\",\"offset\":\"0\",\"limit\":\"200\"},\"header\":{},\"label\":\"name\",\"value\":\"id\",\"body\":{},\"responseUrl\":\"environments\",\"linkedFields\":[\"categories\"],\"remote\":false,\"remoteName\":\"\",\"remoteQueryField\":\"params\"}},\"isUpdateChildAll\":true,\"remote\":\"\",\"allow-clear\":false,\"showPlaceholder\":\"\",\"filter\":\"\",\"linkText\":\"Create environment\",\"linkHref\":\"${origin}/cicd/project/${project_id}/environment?v=1\",\"herfTips\":\"\",\"isSubPipelineTag\":false,\"disabled\":{\"type\":\"fixed\",\"conditions\":[],\"value\":false}},\"category\":\"form\"}]},{\"componentName\":\"d-form-item\",\"id\":1679043833875,\"category\":\"form\",\"props\":{\"label\":\"Artifact Path\",\"field\":\"artifact\",\"help-tips\":\"\",\"visible\":{\"type\":\"fixed\",\"conditions\":[],\"value\":true},\"disabled\":false,\"rules\":[{\"required\":true,\"message\":\"This field cannot be left blank\",\"trigger\":[\"blur\",\"change\"]},{\"regex\":\"^[\\\\s\\\\S]{1,500}$\",\"message\":\"It can contain a maximum of 500 characters.\",\"trigger\":[\"blur\",\"change\"]}]},\"children\":[{\"componentName\":\"d-input\",\"id\":1679043833874,\"props\":{\"inputType\":\"normal\",\"placeholder\":\"请输入\",\"isTrim\":true,\"autocomplete\":\"off\",\"defaultValue\":\"\",\"disabled\":{\"type\":\"fixed\",\"conditions\":[],\"value\":false}},\"category\":\"form\"}]}]}]",
      "front_data_prop": null,
      "label": null,
      "description": null,
      "required": null,
      "extend_prop": {
        "api_options": null,
        "api_type": null,
        "show_placeholder": null,
        "options": null,
        "disabled_conditions": null,
        "visible_conditions": null
      },
      "validation": {
        "required_message": null,
        "regex": null,
        "regex_message": null,
        "max_length": null,
        "min_length": null
      }
    },
    {
      "unique_id": "464bde41709d4b41a59a186ab47bfd47",
      "name": "env_id",
      "default_value": "",
      "plugin_name": "official_microservice_release",
      "version": "0.1.0",
      "type": "link-select",
      "workspace_id": "*",
      "layout_content": "",
      "front_data_prop": null,
      "label": null,
      "description": null,
      "required": null,
      "extend_prop": {
        "api_options": null,
        "api_type": null,
        "show_placeholder": null,
        "options": null,
        "disabled_conditions": null,
        "visible_conditions": null
      },
      "validation": {
        "required_message": null,
        "regex": null,
        "regex_message": null,
        "max_length": null,
        "min_length": null
      }
    },
    {
      "unique_id": "1fa1728f97184c5ba8a49205178f2b80",
      "name": "artifact",
      "default_value": "",
      "plugin_name": "official_microservice_release",
      "version": "0.1.0",
      "type": "d-input",
      "workspace_id": "*",
      "layout_content": "",
      "front_data_prop": null,
      "label": null,
      "description": null,
      "required": null,
      "extend_prop": {
        "api_options": null,
        "api_type": null,
        "show_placeholder": null,
        "options": null,
        "disabled_conditions": null,
        "visible_conditions": null
      },
      "validation": {
        "required_message": null,
        "regex": null,
        "regex_message": null,
        "max_length": null,
        "min_length": null
      }
    }
  ]
}